### PR TITLE
d_a_obj_seat_sword OK

### PR DESCRIPTION
**Pull request by elijah-thomas774**

### CODE REVIEW

File: src/REL/d/a/obj/d_a_obj_seat_sword.cpp [hunk 1]
```diff
@@ -1,13 +1,258 @@
...
+    const dAcPy_c *player = dAcPy_c::LINK;
+
+    mEffPos.x = sEffectPos.x;
+    mEffPos.y = sEffectPos.y;
+    mEffPos.z = sEffectPos.z;
+    mEffPos.rotY(player->rotation.y);
+
+    mEffPos += player->position;
+    mEffPos.y = player->position.y;
+    mEff.fn_80029A10(PARTICLE_RESOURCE_ID_MAPPING_76_, &mEffPos, &rotation, &mScale, nullptr, nullptr);
```
Comment (elijah-thomas774) [inline src/REL/d/a/obj/d_a_obj_seat_sword.cpp:232]: the ordering here is really odd

File: src/REL/d/a/obj/d_a_obj_seat_sword.cpp [hunk 2]
```diff
@@ -1,13 +1,258 @@
...
+        }
+    }
+
+    return SUCCEEDED;
+}
+
+void dAcOSeatSword_c::doInteraction(s32 param0) {
+    if (param0 == 5) {
+        mField_0x7EA = true;
+        Event evt((char *)sEventName, 0x190, 0x100001, nullptr, nullptr);
```
Comment (elijah-thomas774) [inline src/REL/d/a/obj/d_a_obj_seat_sword.cpp:159]: the event flag is not a volatile one! #70

**Comment by robojumper** [comment]
Got it :)

**Comment by elijah-thomas774** [comment]
I guess this means to not code while tired xD

**Pull request merged**